connect/runtime/src/test/java/org/apache/kafka/connect/runtime/distributed/ConnectAssignorTest.java

```diff
@@ -0,0 +1,138 @@
+    public void testAssignConnectorsWhenBalanced() {
+        int num = 2;
+        List<WorkerLoad> existingAssignment = IntStream.range(0, 3)
+                .mapToObj(i -> workerLoad("worker" + i, i * num, num, i * num, num))
+                .collect(Collectors.toList());
+
+        List<WorkerLoad> expectedAssignment = existingAssignment.stream()
+                .map(wl -> WorkerLoad.copy(wl.worker(), wl.connectors(), wl.tasks()))
+                .collect(Collectors.toList());
+        expectedAssignment.get(0).connectors().addAll(Arrays.asList("connector6", "connector9"));
+        expectedAssignment.get(1).connectors().addAll(Arrays.asList("connector7", "connector10"));
+        expectedAssignment.get(2).connectors().addAll(Arrays.asList("connector8"));
+
+        List<String> newConnectors = newConnectors(6, 11);
+        assignor.assignConnectors(existingAssignment, newConnectors);
+        assertEquals(expectedAssignment, existingAssignment);
+    }
+
+    @Test
+    public void testAssignTasksWhenBalanced() {
+        int num = 2;
+        List<WorkerLoad> existingAssignment = IntStream.range(0, 3)
+                .mapToObj(i -> workerLoad("worker" + i, i * num, num, i * num, num))
+                .collect(Collectors.toList());
+
+        List<WorkerLoad> expectedAssignment = existingAssignment.stream()
+                .map(wl -> WorkerLoad.copy(wl.worker(), wl.connectors(), wl.tasks()))
+                .collect(Collectors.toList());
+
+        expectedAssignment.get(0).connectors().addAll(Arrays.asList("connector6", "connector9"));
+        expectedAssignment.get(1).connectors().addAll(Arrays.asList("connector7", "connector10"));
+        expectedAssignment.get(2).connectors().addAll(Arrays.asList("connector8"));
+
+        expectedAssignment.get(0).tasks().addAll(Arrays.asList(new ConnectorTaskId("task", 6), new ConnectorTaskId("task", 9)));
+        expectedAssignment.get(1).tasks().addAll(Arrays.asList(new ConnectorTaskId("task", 7), new ConnectorTaskId("task", 10)));
+        expectedAssignment.get(2).tasks().addAll(Arrays.asList(new ConnectorTaskId("task", 8)));
+
+        List<String> newConnectors = newConnectors(6, 11);
+        assignor.assignConnectors(existingAssignment, newConnectors);
+        List<ConnectorTaskId> newTasks = newTasks(6, 11);
+        assignor.assignTasks(existingAssignment, newTasks);
+        assertEquals(expectedAssignment, existingAssignment);
+    }
+
+    @Test
+    public void testAssignConnectorsWhenImbalanced() {
+        int num = 2;
+        IntUnaryOperator arithmeticProgression = i -> i * ((i - 1) + 2 * num) / 2;
+        List<WorkerLoad> existingAssignment = IntStream.range(0, 3)
+                .mapToObj(i -> workerLoad("worker" + i,
+                        arithmeticProgression.applyAsInt(i), num + i,
+                        arithmeticProgression.applyAsInt(i), num + i))
+                .collect(Collectors.toList());
+        existingAssignment.add(emptyWorkerLoad("worker4"));
```

Comment:
Do you think it's easier to understand the test to simply do:
```
List<WorkerLoad> existingAssignment = new ArrayList<>();
existingAssignment.add(workerLoad("worker0", 0, 1));
existingAssignment.add(workerLoad("worker1", 2, 3));
existingAssignment.add(workerLoad("worker2", 5, 6));
existingAssignment.add(workerLoad("worker3", 9, 10));
existingAssignment.add(emptyWorkerLoad("worker4"));
```

Reply:
I like to keep the programmatic way of defining test cases. 
But you are right in that the workers here won't scale significantlly. 
Changed.